plugins/outputs/whatap/DataOutputX.go

```diff
@@ -0,0 +1,204 @@
+package whatap
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+)
+
+const (
+	INT3_MIN_VALUE  = -8388608 /*0xff800000*/
+	INT3_MAX_VALUE  = 0x007fffff
+	LONG5_MIN_VALUE = -549755813888 /*0xffffff8000000000*/
+	LONG5_MAX_VALUE = 0x0000007fffffffff
+)
+
+type DataOutputX struct {
+	buffer  bytes.Buffer
+	written int
+}
+
+func NewDataOutputX() *DataOutputX {
+	in := new(DataOutputX)
+	return in
+}
```

Comment:
I think you should rather place that one-line body on the caller site.

Reply:
Moved to a library outpside of telegraf